File: data/derived_cot/rq1_traces/translation/qwen/cpp_to_py/code_output/PersonRequest.py
class PersonRequest:
    def __init__(self, name, sex, phone_number):
        self.name = self._validate_name(name)
        self.sex = self._validate_sex(sex)
        self.phone_number = self._validate_phone_number(phone_number)

    def _validate_name(self, name):
        if not name or len(name) > 33:
            return ""
        return name

    def _validate_sex(self, sex):
        if sex not in ["Man", "Woman", "UGM"]:
            return ""
        return sex

    def _validate_phone_number(self, phone_number):
        if not phone_number or len(phone_number) != 11 or not phone_number.isdigit():
            return ""
        return phone_number

    def _is_all_digits(self, s):
        return s.isdigit()

File: data/derived_cot/rq1_traces/translation/qwen/py_to_cpp/code_output/PersonRequest.cpp
#include <string>

class PersonRequest {
private:
    static bool is_valid_sex(const std::string& sex) {
        return sex == "Man" || sex == "Woman" || sex == "UGM";
    }

    static bool is_valid_phone_number(const std::string& phone_number) {
        if (phone_number.empty()) return false;
        if (phone_number.length() != 11) return false;
        for (char c : phone_number) {
            if (!isdigit(static_cast<unsigned char>(c))) 
                return false;
        }
        return true;
    }

public:
    PersonRequest(const std::string& name, const std::string& sex, const std::string& phone_number) 
        : name(_validate_name(name)), 
          sex(_validate_sex(sex)), 
          phone_number(_validate_phone_number(phone_number)) {}

    const std::string& get_name() const { return name; }
    const std::string& get_sex() const { return sex; }
    const std::string& get_phone_number() const { return phone_number; }

private:
    std::string _validate_name(const std::string& name) {
        if (name.empty() || name.length() > 33) 
            return "";
        return name;
    }

    std::string _validate_sex(const std::string& sex) {
        if (!is_valid_sex(sex)) 
            return "";
        return sex;
    }

    std::string _validate_phone_number(const std::string& phone_number) {
        if (!is_valid_phone_number(phone_number)) 
            return "";
        return phone_number;
    }

    std::string name;
    std::string sex;
    std::string phone_number;
};